src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -356,91 +377,111 @@ private void buildSectionPanelsForViewByParticipantQuestionParticipant(
         for (Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> primaryToSecondaryParticipantToResponsesMap
                 : sortedResponses.entrySet()) {
             primaryParticipantIndex += 1;
             String primaryParticipantIdentifier = primaryToSecondaryParticipantToResponsesMap.getKey();
-            
+           
             String currentTeam = getCurrentTeam(bundle, primaryParticipantIdentifier);
             String currentSection = getCurrentSection(primaryToSecondaryParticipantToResponsesMap, viewType);
-            
+           
             boolean isDifferentTeam = !prevTeam.equals(currentTeam);
             boolean isDifferentSection = !prevSection.equals(currentSection);
 
             if (isDifferentTeam) {
                 boolean isFirstTeam = prevTeam.isEmpty();
-                if (!isFirstTeam) {
+                if (!isFirstTeam && isGroupedByTeam()) {
                     // construct missing participant panels for the previous team
-                    buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, teamMembersWithResponses);
-                    teamMembersWithResponses.clear();
+                    buildMissingParticipantPanelsForTeam(
+                            sectionPanel, prevTeam, participantsWithResponses);
+                    participantsWithResponses.clear();
                 }
-                
+               
                 teamsWithResponses.add(currentTeam);
             }
-            
+           
             if (isDifferentSection) {
                 boolean isFirstSection = sectionPanel.getParticipantPanels().isEmpty();
                 if (!isFirstSection) {
                     // Finalize building of section panel,
                     finalizeBuildingSectionPanel(sectionPanel, prevSection, responsesGroupedByTeam, teamsWithResponses);
-                    buildMissingTeamAndParticipantPanelsForSection(
-                            sectionPanel, prevSection, teamsWithResponses);
-                    
-                    // add to sectionPanels,
+                   
+                    // construct missing participant panels for the previous section
+                    if (isGroupedByTeam()) {
+                        buildMissingTeamAndParticipantPanelsForSection(
+                                                        sectionPanel, prevSection, teamsWithResponses);
+                    } else {
+                        buildMissingParticipantPanelsForSection(sectionPanel, prevSection, participantsWithResponses);
+                        participantsWithResponses.clear();
+                    }
+                   
+                   // add to sectionPanels,
                     sectionPanels.put(prevSection, sectionPanel);
                     sectionsWithResponses.add(prevSection);
-                    
-                    // setup for next section
+                   
+                   // setup for next section
```

Comment:
This must be one of the places where the indentation was off by one space before, and fixed during Checkstyle rule enforcement. The comment now is back at its wrong indentation, it'll be great if you can fix it.